internal/tfimport/importer/resources.go

```diff
@@ -28,3 +49,81 @@ func fromConfigValues(key string, cvs ...ProviderConfigMap) (interface{}, error)
 	}
 	return nil, fmt.Errorf("could not find key %q in resource change or provider config", key)
 }
+
+// userValue asks the user to fill in a value that the importer can't figure out.
+func fromUser(in io.Reader, fieldName string, prompt string, choices []string) (val string, err error) {
+	log.Printf("Could not determine %q automatically\n", fieldName)
+	log.Println(prompt)
+
+	if len(choices) > 0 {
+		val, err = userChoice(in, choices)
+	} else {
+		val, err = userValue(in)
+	}
+
+	if val == "" {
+		ie := &InsufficientInfoErr{missingFields: []string{fieldName}}
+		if err != nil {
+			ie.msg = err.Error()
+		}
+		return "", ie
+	}
+	return val, err
+}
+
+func userValue(in io.Reader) (string, error) {
+	fmt.Println("Enter a value manually (blank to skip):")
+	reader := bufio.NewReader(in)
+	val, err := reader.ReadString('\n')
+	if err != nil {
+		return "", err
+	}
+	val = strings.TrimSpace(val)
+	return val, nil
+}
+
+// userChoice asks the user to select one of the possible choices by index.
+// Returns skip==true if the user decided not to make a choice.
+func userChoice(in io.Reader, choices []string) (string, error) {
```

Comment:
Do we currently have a use case for this?

Reply:
Not yet, but I think we will eventually. I'd like to keep it for a little bit until I go through more of the complicated resources, and if we really don't need it, I'll remove it.

Reply:
sg